index.js

```diff
@@ -67,24 +71,32 @@ var template = HandleBars.compile(`
 		</div>
 	</body>
 </html>
-`)
+`);
 
-var payload = {
+const payload = {
 	customerName: config.customerName,
-	taxAmount: taxAmount,
-	amount: amount,
-	cost: cost
+	taxAmount: budget.taxAmount,
+	amount: budget.amount,
+	cost: budget.cost
 }
-
-var output = template(payload);
-var fs = require('fs');
+shine('Rendering template...');
+const output = template(payload);
 
 // generate file
 console.log("budget created, generating file...")
-var stream = fs.createWriteStream("budget.html");
-stream.once('open', () => {
-	stream.write(output);
-	stream.end();
-})
+try {
+	fs.writeFileSync(options.output + "/budget.html", output);
+} catch(e) {
+	fs.mkdirSync(options.output, '0777', (err) => {
+		console.error(err);
+		throw new Error('Could not create output directory : ' )
+		
+	});
+
+	fs.writeFileSync(options.output + "/budget.html", output);
```

Comment:
Maybe the exception catch here are not exactly the error about directory not exists. The second call to `writeFileSync` call not have any benefits being in try catch block. What I suggest here is to have together in `try` block a call to IO to check if file exists and another to create the file, and if an exception occurs you could print exception in console and raise for up.